path.go

```diff
@@ -15,11 +15,76 @@
 package zetcd
 
 import (
-	"path"
+	"fmt"
 )
 
+// validatePath checks the supplied path against the ZK rules for valid node
+// paths. Implements upstream ZK checks from
+// ./src/java/main/org/apache/zookeeper/common/PathUtils.java
+// isSequential modifies the check to see if the path is valid *after* a number
+// is appended.
+func validatePath(zkPath string, isSequential bool) error {
+	if isSequential {
+		zkPath = fmt.Sprintf("%s1", zkPath)
+	}
+
+	if len(zkPath) == 0 {
+		return fmt.Errorf("path cannot be null or zero-length")
```

Comment:
Possibly these should become error types instead to deal with the path handling lower in the function?